src/structure_dump.cpp

```diff
@@ -85,36 +87,60 @@ void structure::dump_chunk_layout(const char *filename) {
   delete[] nums;
 }
 
-void structure::dump(const char *filename) {
-  if (verbosity > 0) master_printf("creating epsilon output file \"%s\"...\n", filename);
-
-  // make/save a num_chunks x NUM_FIELD_COMPONENTS x 5 array counting
-  // the number of entries in the chi1inv array for each chunk.
-  size_t *num_chi1inv_ = new size_t[num_chunks * NUM_FIELD_COMPONENTS * 5];
-  memset(num_chi1inv_, 0, sizeof(size_t) * size_t(num_chunks * NUM_FIELD_COMPONENTS * 5));
+void structure::dump(const char *filename, bool single_parallel_file) {
+  if (verbosity > 0) printf("creating epsilon from file \"%s\" (%d)...\n", filename, single_parallel_file);
```

Comment:
Shouldn't `printf`  be `master_printf`?

Reply:
I wanted this to be a "printf" because I wanted all the processes to log this message since the filename would be different for each of them.